spec/formatting.md

```diff
@@ -19,6 +19,199 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+## Pattern Selection
+
+When formatting a message with one or more _selectors_,
+the _pattern_ of one of the _variants_ must be selected for formatting.
+
+When a message has a single _selector_,
+an implementation-defined method compares each key to the _selector_
+and determines which of the keys match, and in what order of preference.
+A catch-all key will always match, but is always the least preferred choice.
+During selection, the _variant_ with the best-matching key is selected.
+
+In a message with more than one _selector_,
```

Comment:
The intent with these paragraphs is to provide an easier-to-understand but accurate representation of the selection method than what's detailed in the subsequent sections. The split into two cases (N=1, then N>1) is intended to first explain the more common and more straightforward situation, before extending to the more complex.

> I think it is clearer to think of a _selector_ as a function. Then one can describe single-select and the later mulitple-select more generically.

To me, this doesn't quite match with our current [definition](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md#selectors) of the word:
> A **_selector_** is a statement containing one or more expressions which will be used to choose one of the _variants_ during formatting.

If accepted, this is modified by PR #371 to read:
> A `match` statement contains one or more **_selectors_** which will be used to choose one of the _variants_ during formatting.

I read this as describing a _selector_ primarily as a data structure rather than a function, which means that until we actually define such a function or other construct, we can't use it for further explanations.

> I think the single-vs-multiple is a red herring as well: they should work identically.

They do work identically; it's just that with only one we don't need to combine multiple preference orders, which is the main source of our complexity here.